comms/dht/src/connectivity/mod.rs

```diff
@@ -786,68 +796,52 @@ impl DhtConnectivity {
         // - it has the required features
         // - it didn't recently fail to connect, and
         // - it is not in the exclusion list in closest_request
-        let mut connect_ineligable_count = 0;
-        let mut banned_count = 0;
-        let mut excluded_count = 0;
-        let mut filtered_out_node_count = 0;
-        let mut already_connected = 0;
+        // let mut connect_ineligable_count = 0;
+        // let mut banned_count = 0;
+        // let mut excluded_count = 0;
+        // let mut filtered_out_node_count = 0;
+        // let mut already_connected = 0;
+        let offline_cooldown = self.config.offline_peer_cooldown;
         let query = PeerQuery::new()
             .select_where(|peer| {
                 if peer.is_banned() {
-                    banned_count += 1;
+                    // banned_count += 1;
                     return false;
                 }
 
                 if peer.features.is_client() {
-                    filtered_out_node_count += 1;
+                    // filtered_out_node_count += 1;
                     return false;
                 }
 
                 if connected.contains(&&peer.node_id) {
-                    already_connected += 1;
+                    // already_connected += 1;
                     return false;
                 }
 
                 if peer
                     .offline_since()
-                    .map(|since| since <= self.config.offline_peer_cooldown)
+                    .map(|since| since <= offline_cooldown)
                     .unwrap_or(false)
                 {
-                    connect_ineligable_count += 1;
+                    // connect_ineligable_count += 1;
                     return false;
                 }
 
                 let is_excluded = excluded.contains(&peer.node_id);
                 if is_excluded {
-                    excluded_count += 1;
+                    // excluded_count += 1;
                     return false;
                 }
 
                 true
             })
-            .sort_by(PeerQuerySortBy::DistanceFromLastConnected(node_id))
-            // Fetch double here so that there is a bigger closest peer set that can be ordered by last seen
-            .limit(n * 2);
+            // TODO: Check if this logic is correct. At the moment I want them in the same order every time
```

Comment:
My problem with the previous approach was that it could end up in a stale loop, if the peers came online again before new peers were tried. This approach is more predictable and ensures that all the peers are tried. It may be a symptom of the current network state, but the majority of these nodes are actually offline. 

Given that the offline data is more accurate now, it may be much of a muchness. I think I would however lobby for selecting the same n (instead of n*2) peers from the DB and returning them directly.

Reply:
`n == 8` by default. 
The goal was to fetch peers that we recently connected to, ordered by distance. When a peer comes back online, it will send a join message, which will be propagated to neighbours, and/or directly connect. This will update the last_seen which will put them back on the top of the list for dialing if ordered by last seen. 

This change will select the same 8 peers (unless a nearer one joins) which may never come online again. Currently "culling" of offline peers is a slow process, I think we have to attempt and fail to connect to them at least once and then we have to have not connected to them for 24 hours before we remove them from the peer list entirely. 

I think a good solution would be to maintain a list that can be loaded that has available peers. Essentially just optimising this peer query from an LMDB perspective.

Reply:
In this PR, as soon as an address fails it is immediately marked offline. If all addresses are offline, the peer is offline